source/PlayerInfo.cpp

```diff
@@ -1346,6 +1346,17 @@ void PlayerInfo::ReorderShip(int fromIndex, int toIndex)
 	shared_ptr<Ship> ship = ships[fromIndex];
 	ships.erase(ships.begin() + fromIndex);
 	ships.insert(ships.begin() + toIndex, ship);
+	// Check if the ship in the first position can be a flagship and is in the current system.
+	auto firstShipIt = ships.begin();
+	if(Preferences::Has("Automatically unpark flagship")
+			&& (*firstShipIt)->CanBeFlagship() && (*firstShipIt)->GetSystem() == system)
+	{
+		if(fromIndex == 0)
+			ships[toIndex]->SetIsParked(true);
+		else
+			ships[fromIndex]->SetIsParked(true);
+		(*firstShipIt)->SetIsParked(false);
+	}
```

Comment:
```suggestion
	// Check if the ship in the first position can be a flagship and is in the current system.
	auto firstShip = ships.begin();
	if(Preferences::Has("Automatically unpark flagship")
			&& (*firstShip)->CanBeFlagship() && (*firstShip)->GetSystem() == system)
	{
		if(fromIndex == 0)
			ships[toIndex]->SetIsParked(true);
		else
			ships[fromIndex]->SetIsParked(true);
		(*firstShip)->SetIsParked(false);
	}
```
`firstshipIt` doesn't add anything to the code, just `firstShip` is fine.